src/components/pages/WelcomePage.tsx

```diff
@@ -185,6 +183,22 @@ class WelcomePage extends React.Component<Props, State> {
         linkedEventsPost: linkedEventsPost
       });
     });
+  
+    await this.getPopularPagesImageUrl();
+  }
+
+  /**
+   * Fetch PopularPages featured image URL 
+   */
+  private getPopularPagesImageUrl = () => {
+    const { popularPages } = this.state;
+    return popularPages.map(async(page) => {
+      const pageId = page.id ? page.id.toString() : "";
+      const popularImageUrl = await ApiUtils.getApi().getPostThumbnail({ id: pageId });
+      this.setState({
+        featureImageUrl: [...this.state.featureImageUrl, popularImageUrl]
+      })
+    });
```

Comment:
Consider using a variable to collect all items in the array and putting that to the this.setState(). Current solution will render dom tree again each iteration.

Example:
`
const array1 = [1, 2];
const array2 = array1.map((item) => {
      return `${item}. item`;
});
this.setState({
      strArray: array2
});
`